akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ConsistentHashingShardAllocationStrategy.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding
+
+import scala.collection.immutable
+import scala.concurrent.Future
+
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import akka.actor.Address
+import akka.cluster.Cluster
+import akka.cluster.ClusterEvent.CurrentClusterState
+import akka.cluster.Member
+import akka.cluster.sharding.ShardCoordinator.ActorSystemDependentAllocationStrategy
+import akka.cluster.sharding.ShardRegion.ShardId
+import akka.cluster.sharding.internal.ClusterShardAllocationMixin
+import akka.cluster.sharding.internal.ClusterShardAllocationMixin.ShardSuitabilityOrdering
+import akka.routing.ConsistentHash
+
+object ConsistentHashingShardAllocationStrategy {
+  private val emptyRebalanceResult = Future.successful(Set.empty[ShardId])
+}
+
+/**
+ * [[akka.cluster.sharding.ShardCoordinator.ShardAllocationStrategy]] that is using consistent
+ * hashing. This can be useful when shards with the same shard id for different entity types
+ * should be best effort collocated to the same nodes.
```

Comment:
> ShardedDaemonProcess is using the process index as the shard id, which is a perfect fit for this (process index is the selector of slice range)

@johanandren can you confirm that this is correct (so I didn't misread the code)

Reply:
Sounds right, encoded sdp id is `[revision]|[total-size]|[id]` and we extract the last as shard.